src/OSmOSE/cluster/audio_reshaper.py

```diff
@@ -1,630 +1,355 @@
-
-            if not skip_last:
-                outfilename = output_dir_path.joinpath(
-                    f"{datetime.strftime(timestamp, '%Y-%m-%dT%H:%M:%S').replace('-','_').replace(':','_').replace('.','_').replace('+','_')}.{fmt}"
-                )
-                result.append(outfilename.name)
-
-                timestamp_list.append(
-                    datetime.strftime(timestamp, "%Y-%m-%dT%H:%M:%S.%f%z")
-                )
-                timestamp += timedelta(seconds=len(output))
-
-                sf.write(
-                    outfilename, output, sample_rate, format=fmt, subtype=file_type
+                    f"File {outfilename} already exists and overwrite is set to False. Skipping...\n"
                 )
-                os.chmod(outfilename, mode=FPDEFAULT)
-
-                if verbose:
-                    print(
-                        f"{outfilename} written! File is {(len(output)/sample_rate)} seconds long. {len(previous_audio_data)/sample_rate} seconds left from slicing."
-                    )
+            continue
 
-        # in particular, it is here that we delete files that have been resampled and copied in outputdir
-        for remaining_file in [f for f in files if input_dir_path.joinpath(f).exists()]:
-            if (
-                overwrite
-                and not implicit_output
-                and output_dir_path == input_dir_path
-                and last_file_behavior != "discard"
-            ):
-                print(f"Deleting {remaining_file}")
-                input_dir_path.joinpath(remaining_file).unlink()
+        result.append(outfilename.name)
+        timestamp_list.append(segment_datetime_begin.strftime("%Y-%m-%dT%H:%M:%S.%f%z"))
 
-        input_timestamp = pd.DataFrame(
-            {"filename": result, "timestamp": timestamp_list}
-        )
-        input_timestamp.sort_values(by=["timestamp"], inplace=True)
-        input_timestamp.drop_duplicates().to_csv(
-            output_dir_path.joinpath(f"timestamp_{batch_ind_min}.csv"),
-            index=False,
-            na_rep="NaN",
-        )
-        os.chmod(
-            output_dir_path.joinpath(f"timestamp_{batch_ind_min}.csv"), mode=FPDEFAULT
+        sf.write(
+            outfilename,
+            audio_data,
+            segment_sample_rate,
         )
+        os.chmod(outfilename, mode=FPDEFAULT)
 
-        # path_csv = output_dir_path.joinpath("timestamp.csv")
-        # lock = FileLock(str(path_csv) + ".lock")
+        if verbose:
+            print(
+                f"Saved file from {segment_datetime_begin} to {segment_datetime_end} as {outfilename}\n"
+            )
 
-        # with lock:
-        #     # suppr doublons
-        #     if path_csv.exists():
-        #         tmp_timestamp = pd.read_csv(path_csv)
-        #         result += list(tmp_timestamp["filename"].values)
-        #         timestamp_list += list(tmp_timestamp["timestamp"].values)
+    # writing infos to timestamp_*.csv
```

Comment:
I guess lots of stuff like this could benefit from being put in specific functions